### PR TITLE
stm32mp1: update documentation on build directives

### PR DESCRIPTION
**Pull request by etienne-lms**

### CODE REVIEW

File: building/devices/stm32mp1.rst [hunk 1]
```diff
@@ -45,14 +45,14 @@ as listed in table below:
...
 +------------------------+--------------------------------------+
-| `STM32MP157C-DK2`      | ``PLATFORM=stm32mp1-157C_DK2``       |
-| `STM32MP157F-DK2`      |                                      |
+| `STM32MP157C-DK2`_     | ``PLATFORM=stm32mp1-157C_DK2_SCMI``  |
+| `STM32MP157F-DK2`_     |                                      |
 +------------------------+--------------------------------------+
-| `STM32MP157C-EV1`      | ``PLATFORM=stm32mp1-157C_EV1``       |
-| `STM32MP157F-EV1`      |                                      |
+| `STM32MP157C-EV1`_     | ``PLATFORM=stm32mp1-157C_EV1_SCMI``  |
+| `STM32MP157F-EV1`_     |                                      |
```
Comment (GseoC) [inline building/devices/stm32mp1.rst:55]: I think we should let the non-SCMI variant appear as it is still supported

**Comment by GseoC** [comment]
For commits:
"stm32mp1: fix references formatting"
"stm32mp1: update documentation on build directives"
"platform: enumerate the few stm32mp1 supported boards"

`Reviewed-by: Gatien Chevallier <gatien.chevallier@foss.st.com>`

**Comment by jbech-linaro** [comment]
`Reviewed-by: Joakim Bech <joakim.bech@linaro.org>`

File: building/devices/stm32mp1.rst (hunk 1 above)
Comment (etienne-lms) [reply building/devices/stm32mp1.rst:55]: Ok, i'll add them back...

**Comment by GseoC** [comment]
`Reviewed-by: Gatien Chevallier <gatien.chevallier@foss.st.com>`

**Comment by etienne-lms** [comment]
Fixup commit squashed and review tags applied.

**Pull request merged**